banyand/stream/block.go

```diff
@@ -543,12 +548,30 @@ func (bc *blockCursor) loadData(tmpBlock *block) bool {
 	bc.bm.tagFamilies = tf
 	tmpBlock.mustReadFrom(&bc.tagValuesDecoder, bc.p, bc.bm)
 
-	start, end, ok := timestamp.FindRange(tmpBlock.timestamps, bc.minTimestamp, bc.maxTimestamp)
-	if !ok {
-		return false
+	idxList := make([]int, 0)
+	var start, end int
+	if applyFilter {
```

Comment:
The "applyFilter" is unnecessary. Instead, you can simply check if "expectedTimestamps" is equal to nil.